tt_metal/distributed/mesh_device.cpp

```
@@ -887,6 +890,9 @@ bool MeshDevice::initialize(
     size_t /*worker_l1_size*/,
     tt::stl::Span<const std::uint32_t> /*l1_bank_remap*/,
     bool /*minimal*/) {
+    if (this->is_initialized()) {
```

How hard is it to make it a fatal?

Properly there are only two options.:
- check the contract (can’t be called twice)
- allow (by resetting the state)

Here is makes the most sense to go with a contract check. 

I don’t like to silently return here because here method does not follow the contract. User might call it the second time with different params, method will act like init is done but in reality it’s not.

---

Yes, agreed - this is better. I've adjusted it to `TT_FATAL` and added a unit test.